### ISSUE TITLE
Updated hardware kit or BOM?

### ISSUE THREAD

**Issue by ghost** [f-of-e/issues#1]
Is there an updated version of the hardware kit to account for recent releases?

The Raspberry Pi Pico is a new microcontroller with the Cortex M0+ processor. 

The UPduino v3.0 is a low cost Lattice iCE40 FPGA board.

Could these be used for the exercises on this course?

**Comment by phillipstanleymarbell** [comment]
Yes, you are right, both the Raspberry Pi Pico and the UPduino would be good candidates to replace the current hardware used.

For the FPGA component, the existing tools (https://github.com/f-of-e/f-of-e-tools) should work on the UPduino. **I would much appreciate a pull request if you have an UPduino hardware and can test it**!

For the microcontroller component, the goal is to very soon migrate all the Cortex-M0 component of the course to a RISC-V part. One challenge is that, for the version of the course at the University of Cambridge, we have hardware kits which we loan to students each year, so the process of switching to new hardware requires a bit more planning  (and cost) in addition to updating the material.

**Comment by ghost** [comment]
Brilliant! I’ll add my findings to that tools repo.

Would be useful to have a few community recommendations based on how easy it is to find the boards / how good the boards are. 

I was doing a bit of research on FPGA boards the other day and there are so many choice 😀.

**Comment by ghost** [comment]
Also realised that it may be possible to use AWS instances for FPGA and RISC-V development. 

Would be useful to highlight the best images to use, setup, etc.

**Comment by phillipstanleymarbell** [comment]
> Also realised that it may be possible to use AWS instances for FPGA and RISC-V development.
> Would be useful to highlight the best images to use, setup, etc.

Yes, indeed, good point. For the Cambridge version of the course, I provide students access to an AWS Ubuntu server instance with the tools preinstalled.

https://f-of-e.github.io/HtmlChapters/appendix-GB3-setup/appendix-GB3-setup.html describes how to setup the FPGA tools on a Linux system. I have disabled the corresponding component and videos on the F-of-E site (Fascicle 03) since I am doing some cleanup before re-releasing the FPGA part of the material (for a course that runs in May/June). Hope that helps for now.
